dcicutils/misc_utils.py

```diff
@@ -412,6 +416,68 @@ def utc_today_str():
     return datetime.datetime.strftime(datetime.datetime.utcnow(), "%Y-%m-%d")
 
 
+def as_seconds(*, seconds=0, minutes=0, hours=0, days=0, weeks=0, milliseconds=0, as_type=None):
+    """
+    Coerces a relative amount of time (keyword arguments seconds, minutes, etc. like timedelta) into seconds.
+
+    If the number of seconds is an integer, it will be coerced to an integer. Otherwise, it will be a float.
+    If as_float is given and not None, it will be applied as a function to the result, allowing it to be coerced
+    to another value than an integer or float.  For example,
+      >>> as_seconds(seconds=1, minutes=1)
+      61
+      >>> as_seconds(seconds=1, minutes=1, as_type=str)
+      '61'
+    """
+    delta = datetime.timedelta(seconds=seconds, minutes=minutes, hours=hours,
+                               days=days, weeks=weeks, milliseconds=milliseconds)
+    seconds = delta.total_seconds()
+    frac, intpart = math.modf(seconds)
+    if frac == 0.0:
+        seconds = int(intpart)
+    if as_type is not None:
+        seconds = as_type(seconds)
+    return seconds
+
+
+HMS_TZ = pytz.timezone("US/Eastern")
+
+
+def as_datetime(dt, tz=None):
+    """
+    Parses the given datetime, returning a datetime.datetime object.
+
+    If the given datetime is already such an object, it is returned.
+    """
+    if dt is None:
+        return None
+    try:
+        # This type check has to work even if datetime is mocked, so we use it under another variable name to
+        # make it harder to mock out. -kmp 6-Nov-2020
+        if not isinstance(dt, datetime_type):
+            dt = dateutil_parse(dt)
+        if tz and not dt.tzinfo:
+            dt = tz.localize(dt)
+        return dt
+    except Exception:
+        return None
```

Comment:
It's not clear to me from the definition what the utility of this function is... If it's already a datetime, how is it not a datetime.datetime object? It seems like you're doing all this casting from datetime objects that is unnecessary... If `dt` were a string this would make sense to me...

Reply:
It takes a string as an argument. Accepting a datetime also is just for convenience so that other functions can accept times in either parsed or unparsed form. Look to line 457 where it parse a time string into a datetime.

Reply:
I guess I need to make it clearer in the doc string. To me, anything that uses the verb parses means it takes a string (or I guess sometimes a stream, though not here), so the idea that you would think it only took a datetime didn't enter my mind. :)

Reply:
If you look at the tests, you'll see the intended uses:
```
def test_as_datetime():

    t0 = datetime_module.datetime(2015, 7, 4, 12, 0, 0)
    t0_utc = pytz.UTC.localize(datetime_module.datetime(2015, 7, 4, 12, 0, 0))

    assert as_datetime(t0) == t0
    assert as_datetime(t0, tz=pytz.UTC) == t0_utc
    assert not as_datetime('2015-07-04T12:00:00').tzinfo
    assert as_datetime('2015-07-04T12:00:00') == t0
    assert as_datetime('2015-07-04 12:00:00') == t0
    assert as_datetime('2015-07-04 12:00:00', tz=pytz.UTC) == t0_utc
    assert as_datetime('2015-07-04 12:00:00Z').tzinfo
    assert as_datetime('2015-07-04 12:00:00Z') != t0
    assert as_datetime('2015-07-04 12:00:00Z') == t0_utc
    assert as_datetime('2015-07-04 12:00:00-0000') == t0_utc
```